src/GraphQL/ObjectExtensions.cs

```diff
@@ -78,6 +78,19 @@ public static object GetPropertyValue(this object propertyValue, Type fieldType)
                 var underlyingType = Nullable.GetUnderlyingType(elementType) ?? elementType;
                 var implementsIList = fieldType.GetInterface("IList") != null;
 
+                if (fieldType.IsArray)
+                {
+                    if (!(propertyValue is IList values))
+                    {
+                        return null;
+                    }
+
+                    var tempArray = Array.CreateInstance(elementType, values.Count);
```

Comment:
I think this should also use `underlyingType` vs. `elementType`, since the `elementType` could be nullable.  That would be a good test to add.

Reply:
Hi, 
I've added some unit tests and changed `IList` to `ICollection`.
If we want to have an array: `double?[]`, then I believe taking the `elementType` is the only way as it preserve information about nullablelity. 
With `underlyingType` we would get the `double[]`

Reply:
👍 to using `elementType`.

Is the reason for checking for `ICollection` so that you can use `CopyTo`?  One test that we're missing is one for an array of objects (or an array of arrays).  I don't think you'll be able to use `CopyTo` with that.  So I think that check should just be against `IEnumerable` and then build up the values.

```
var valueList = propertyValue as IEnumerable;
if (valueList == null) return newArray;

valueList.Apply((listItem, idx) =>
{
  tempArray[idx] = listItem == null ? null : GetPropertyValue(listItem, elementType);
});

return tempArray;
```